aten/src/ATen/native/cuda/ForeachUnaryOp.cu

```diff
@@ -70,7 +110,75 @@ void foreach_tensor_##NAME##_cuda_(TensorList tensors) {                \
     foreach_unary_op_<NAME1>(tensors);                                  \
 }
 
+#define FOREACH_UNARY_OP_NO_COMPLEX(NAME, NAME1)                        \
+template<typename T>                                                    \
+struct NAME1 {                                                          \
+    __device__ T operator()(T t) const { return std::NAME(t); }         \
+};                                                                      \
+                                                                        \
+std::vector<Tensor> foreach_tensor_##NAME##_cuda(TensorList tensors) {  \
+    check_foreach_api_restrictions(tensors);                            \
+                                                                        \
+    if (!can_use_fast_route(tensors)) {                                 \
+        return at::native::foreach_tensor_##NAME##_slow(tensors);       \
+    }                                                                   \
+                                                                        \
+    return foreach_unary_op_no_comlex<NAME1>(tensors);                  \
+}                                                                       \
+                                                                        \
+void foreach_tensor_##NAME##_cuda_(TensorList tensors) {                \
+    check_foreach_api_restrictions(tensors);                            \
+                                                                        \
+    if (!can_use_fast_route(tensors)) {                                 \
+        return at::native::foreach_tensor_##NAME##_slow_(tensors);      \
+    }                                                                   \
+                                                                        \
+    foreach_unary_op_no_comlex_<NAME1>(tensors);                        \
+}
+
 FOREACH_UNARY_OP(exp, Exp);
 FOREACH_UNARY_OP(sqrt, Sqrt);
+FOREACH_UNARY_OP(acos, Acos);
+FOREACH_UNARY_OP(asin, Asin);
+FOREACH_UNARY_OP(atan, Atan);
+FOREACH_UNARY_OP(cos, Cos);
```

Comment:
`cos` supports bfloat16 on cuda, for parity mta should also dispatch to bfloat16